upsetplot/data.py

```diff
@@ -16,3 +19,51 @@ def generate_data(seed=0, n_samples=10000, n_sets=3, aggregated=False):
     if aggregated:
         return df.value.groupby(level=list(range(n_sets))).count()
     return df.value
+
+
+def from_sets(data):
+    """
+    Data loader for a dict of sets
+
+    :param Dict[str, set[str]] data:
+    :return: Multi-Index Series of intersection counts
+    :rtype: pd.Series
+    """
+    # Convert dict into OrderedDict to preserve Key/Value order
+    data = OrderedDict(data)
+
+    # Construct Index
+    tf_array = [[True, False]] * len(data)
+    index = pd.MultiIndex.from_product(tf_array, names=data.keys())
```

Comment:
The iteration order of `values()` may not be guaranteed to match that of `keys()`. The order is also non-deterministic across platforms or sessions. You need to use `sorted(data.keys())` (or similar) in order to offer a reliable interface. Please add a test case asserting the order of columns.

Reply:
`data` is cast as an `OrderedDict` on the line above to preserve Key/Value order.